proposals/clone_mulitple_resource.md

```diff
@@ -0,0 +1,163 @@
+# Meta
+[meta]: #meta
+- Name: Allow cloning multiple resources from a Namespace
+- Start Date: 2022-08-04
+- Update data (optional): 2022-08-04
+- Author: [Prateek Pandey](https://github.com/prateekpandey14)
+- Supersedes:  N/A
+
+# Table of Contents
+[table-of-contents]: #table-of-contents
+- [Meta](#meta)
+- [Table of Contents](#table-of-contents)
+- [Overview](#overview)
+- [Definitions](#definitions)
+- [Motivation](#motivation)
+- [Proposal](#proposal)
+- [Implementation](#implementation)
+- [Migration (OPTIONAL)](#migration-optional)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Prior Art](#prior-art)
+- [Unresolved Questions](#unresolved-questions)
+- [CRD Changes (OPTIONAL)](#crd-changes-optional)
+
+# Overview
+[overview]: #overview
+
+Enhance kyverno generate rule to clone mulitple resources from a given namespace.
+
+
+# Definitions
+[definitions]: #definitions
+
+N/A
+
+# Motivation
+[motivation]: #motivation
+
+The initial idea for this feature came from issue [282](https://github.com/kyverno/kyverno/issues/282) and received feedback from the kyverno community in slack.
+
+It would be good to have a way to copy/clone multiple resources from a namespace in a single rule, instead of
+writing a new rule or policy.
+
+There are primarily main use cases associated with this.
+
+1. To clone set of resources from a common namespace, using a match selector whenever we create a new namespace
+   to bootstrap the deployment process.
+
+2. Implicit, User has to less number of policy/rule to copy the set of necessary resources.
+
+# Proposal
+
+In this proposal, Kyverno generate rule can be extended to allowing to clone multiple resources from a namespace.
+This could be done using  matching selector to select multiple resources of the given apiVersion/group and allowing
+multiple kinds/name pairs, similar to the match/exclude blocks we use to filter resources in the admission control
+webhooks.
+
+In below extended ClusterPolicy example, multiple resources can be selected using a kinds list to copy/clone from 
+`stagging` namespace to newly created namespace.
+
+```yaml
+    generate:
+      namespace: "{{request.object.metadata.name}}"
+      synchronize : true
+      cloneList:
+      - kinds:
+         - v1/Secret    // clone resource with matching selector i.e. configmap, secrets etc
+          namespace: stagging
```

Comment:
Still doesn't look right. @prateekpandey14 - can you update this from your local branch?
```
      - kinds:
         - v1/Secret      // clone resource with matching selector i.e. configmap, secrets etc
            namespace: stagging
```